### PR TITLE
Upgrade Parcel V2

### PR DESCRIPTION
**Pull request by sebjacobs**
https://trello.com/c/7NnXpdY6

We recently received a number of Dependabot security warnings [1] caused our dependency on Parcel V1.

This upgrade was a bit fiddly because most of our build dependencies were out-of-date and the fact that Parcel V2 doesn't work on Node V10.

I therefore broke the process up into steps:

* Removing noise caused by newline character differences (469c510)
* Cleaning up unused / miscategorized dependencies (d773a9f...471a9f5)
* Upgrading build dependencies(21c005d...4bfc7db)
* Upgrading Node V14 (38d6d4c)
* Upgrading Parcel V2 (1c85226...18ed4b1)

[1]
https://github.com/futurelearn/markdown-editor/security/dependabot/example/package-lock.json/glob-parent/open
https://github.com/futurelearn/markdown-editor/security/dependabot/example/package-lock.json/is-svg/open
https://github.com/futurelearn/markdown-editor/security/dependabot/example/package-lock.json/node-forge/open

### CODE REVIEW

**Review by kohrVid — COMMENTED** [review]
Thanks for working on this @sebjacobs! I have a few questions regarding the changes made so far so I've left a few comments.
I'm personally having trouble running the Cypress tests for this branch locally (I get the following error: `UnhandledPromiseRejectionWarning: Error: Could not locate the bindings file`) but as the tests seem to pass in CI, I wonder if I there's just an issue with my setup.

File: tsconfig.json [hunk 1]
```diff
@@ -1,31 +1,35 @@
...
-    "noUnusedParameters": true,
+    // linter checks for common issues
     "noImplicitReturns": true,
     "noFallthroughCasesInSwitch": true,
+    // noUnused* overlap with @typescript-eslint/no-unused-vars, can disable if duplicative
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    // use Node's module resolution algorithm, instead of the legacy TS one
     "moduleResolution": "node",
-    "baseUrl": "./",
```
[the remark this answers is not in the document]
Comment (sebjacobs) [reply tsconfig.json:23]: Fixed in b7c93da

File: package.json [hunk 2]
```diff
@@ -66,7 +66,7 @@
     "@types/react-dom": "^16.9.4",
     "cypress": "^3.8.2",
     "cypress-file-upload": "^3.5.3",
-    "node-sass": "^4.13.1",
+    "node-sass": "^6.0.1",
```
[the remark this answers is not in the document]
Comment (sebjacobs) [reply package.json]: Switching from `node-sass` to `sass` appeared to break the sass inline imports transformer provided by `rollup-plugins-postcss`. I spent some time investigating this issue this morning and have come up with a workaround using the `postcss-import` plugin:

0c4103f

**Comment by sebjacobs** [comment]
> Thanks for working on this @sebjacobs! I have a few questions regarding the changes made so far so I've left a few comments.

Thanks for taking a look @kohrVid! I have tweaked the `tsconfig.json` based on your feedback (
b7c93da) and have had another go at switching from `node-sass` to `sass` (0c4103f).

> I'm personally having trouble running the Cypress tests for this branch locally (I get the following error: UnhandledPromiseRejectionWarning: Error: Could not locate the bindings file) but as the tests seem to pass in CI, I wonder if I there's just an issue with my setup.

🤔 I wonder whether there's some additional cache you need to clear after upgrading to Node v14. For example:
* `.cache`
* `node_modules`
* `example/node_modules`
* `example/.parcel-cache`
* or `example/dist`?

**Comment by jamiecobbett** [comment]
I was able to run the tests, but one of the consistently fails for me: `supports multiple code blocks of different languages` with:

> CypressError: Timed out retrying: expected '<input>' to have value '~~~javascript\nvar name = "code";\n~~~\n\n~~~python\ndef name:\n    code\n~~~', but the value was '~~~javascript\nvar name = "code";\n~~~\n\n~~~\ndef name:\n    code\n~~~'

I also have to manually click "Run all tests" in Cypress before anything happens, but this was the case before.

**Comment by sebjacobs** [comment]
@squiddy-bot merge

**Pull request merged**